Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/PanicMod.java

```diff
@@ -1,41 +1,52 @@
-/*
- * Copyright � 2014 - 2016 | Wurst-Imperium | All rights reserved.
- * 
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-package tk.wurst_client.mods;
-
-import tk.wurst_client.events.listeners.UpdateListener;
-import tk.wurst_client.mods.Mod.Category;
-import tk.wurst_client.mods.Mod.Info;
-
-@Info(category = Category.MISC,
-	description = "Instantly turns off all enabled mods.\n"
-		+ "Be careful with this!",
-	name = "Panic",
-	tags = "legit, disable",
-	help = "Mods/Panic")
-public class PanicMod extends Mod implements UpdateListener
-{
-	@Override
-	public void onEnable()
-	{
-		wurst.events.add(UpdateListener.class, this);
-	}
-	
-	@Override
-	public void onUpdate()
-	{
-		for(Mod mod : wurst.mods.getAllMods())
-			if(mod.getCategory() != Category.HIDDEN && mod.isEnabled())
-				mod.setEnabled(false);
-	}
-	
-	@Override
-	public void onDisable()
-	{
-		wurst.events.remove(UpdateListener.class, this);
-	}
-}
+/*
+ * Copyright � 2014 - 2016 | Wurst-Imperium | All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package tk.wurst_client.mods;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import tk.wurst_client.events.listeners.UpdateListener;
+import tk.wurst_client.mods.Mod.Category;
+import tk.wurst_client.mods.Mod.Info;
+
+@Info(category = Category.MISC,
+	description = "Instantly turns off all enabled mods.\n"
+		+ "Be careful with this!",
+	name = "Panic",
+	tags = "legit, disable",
+	help = "Mods/Panic")
+public class PanicMod extends Mod implements UpdateListener
+{
+	
+	private List<Mod> hiddenMods = new ArrayList<>();
+	
+	@Override
+	public void onEnable()
+	{
+		wurst.events.add(UpdateListener.class, this);
+	}
+	
+	@Override
+	public void onUpdate()
+	{
+		for(Mod mod : wurst.mods.getAllMods())
+			if(mod.getCategory() != Category.HIDDEN && mod.isEnabled())
+			{
+				mod.setEnabled(false);
```

Comment:
See #24 for my code... I dunno... It might have those fixes? already?
Should I reopen the pull?

On Sun, May 22, 2016 at 11:11 AM, Mateus Mercer <notifications@github.com>
wrote:

> In Wurst Client for MC 1.9.X/src/tk/wurst_client/mods/PanicMod.java
> <https://github.com/Wurst-Imperium/Wurst-Client-for-MC-1.9.X/pull/25#discussion_r64151281>
> :
>
> > +	
> > +	private List<Mod> hiddenMods = new ArrayList<>();
> > +	
> > +	@Override
> > +	public void onEnable()
> > +	{
> > +		wurst.events.add(UpdateListener.class, this);
> > +	}
> > +	
> > +	@Override
> > +	public void onUpdate()
> > +	{
> > +		for(Mod mod : wurst.mods.getAllMods())
> > +			if(mod.getCategory() != Category.HIDDEN && mod.isEnabled())
> > +			{
> > +				mod.setEnabled(false);
>
> As you can see here, in the first loop it will disable all the mods, so
> the statement above will be always false.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Wurst-Imperium/Wurst-Client-for-MC-1.9.X/pull/25/files/4690c03c1c9fd43321d8961b1a99cb7d2050901c#r64151281>
>



-- 
Regards,
Dhruv Gramopadhye
-----
Controll Freak Operator and Everything
dGRAMOP Development
Penetration Testing | Animation and Rendering | Web Development | Mobile
App Development | Embedded Electronics | and Misc Java Development.
------------------------
http://socnet.16mb.com - My most recent and best project - still buggy,
still work in progress. Feel free to create an account
http://dgramop.co - An index page of my projects
http://m.freeplay.url.ph - A neat looking version of the latter link
http://freeplay.url.ph - An index page of random experiments.
http://ppn.hol.es - A currency system that is online that I use to "bribe"
other children. Over 1.5 years old, I already can name a bunch of security
flaws. Try ../ attack when creating an account.
http://m.ppn.hol.es - An even more neat looking version of the above,
scrolling only works on mobile browsers unless you really zoom in. The junk
text is there to enable scrolling.